cloud_functions/test/flagCryptominer-error-tests.js

```diff
@@ -0,0 +1,88 @@
+// These tests cover various error cases for the flagCryptominer cloud function.
+// Based on https://cloud.google.com/functions/docs/testing/test-background#unit_tests
+//
+// Since this is a background cloud function, errors are exposed via Cloud Logging Error Reporting.
+// Uncaught exceptions will log errors, but will also incur a cold start on future invocations.
+// Instead, we use console.error, so we need to stub that function for these tests.
+// See https://cloud.google.com/functions/docs/monitoring/error-reporting for details.
+//
+// IMPORTANT: Since we're stubbing a global function (console.error), all of these tests must run
+// in serial, not parallel as is Ava's default.
+// (For future exploration: https://github.com/sinonjs/sinon-test)
+const test = require('ava')
+const flagCryptominer = require('../flagCryptominer')
+const sinon = require('sinon')
+const utils = require('./test-utils')
+
+
+test.beforeEach(t => {
+  sinon.stub(console, 'error')
+})
+
+test.afterEach.always(t => {
+  console.error.restore()
+})
+
+
+// The function should do nothing if the cryptominer attribute isn't set. This could indicate a
+// misconfigured Pub/Sub topic subscription.
+test.serial('reject non-cryptominer messages', async t => {
```

Comment:
Is the sole purpose of this flag to cause an error if it's not present? If so, I think it's adding unnecessary complexity.

Reply:
There has been ongoing discussion about the setup of the Pub/Sub topic and subscription. The current plan is to have a general cryptomining topic in `broad-dsde-<env>` which will receive potentially multiple types of messages with multiple publishers and multiple subscribers. This flag is the current agreement for how these messages will be identified. If this code somehow receives a message without this flag, I want the function to do nothing instead of possibly doing the wrong thing.